extensions/amp-disqus/0.1/amp-disqus.js

```diff
@@ -0,0 +1,412 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {CSS} from '../../../build/amp-disqus-0.1.css';
+import {documentInfoFor} from '../../../src/document-info';
+import {generateGUID} from './guid';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {sanitizeHtml} from '../../../src/sanitizer';
+import {user} from '../../../src/log';
+import {xhrFor} from '../../../src/xhr';
+
+
+function timeSince(timeStamp) {
+  const secondsPast = (Date.now() - Date.parse(timeStamp)) / 1000;
+
+  if (secondsPast < 60) {
+    return parseInt(secondsPast, 10) + ' seconds ago';
+  }
+  if (secondsPast < 3600) {
+    return parseInt(secondsPast / 60, 10) + ' minutes ago';
+  }
+  if (secondsPast < 86400) {
+    return parseInt(secondsPast / 3600, 10) + ' hours ago';
+  }
+  if (secondsPast < 2592000) {
+    return parseInt(secondsPast / 86400, 10) + ' days ago';
+  }
+
+  return parseInt(secondsPast / 2592000, 10) + ' months ago';
+}
+
+
+function isDNTEnabled() {
+  return (
+    window.navigator.doNotTrack === '1' ||  // W3 Spec & Firefox
+      window.navigator.doNotTrack === 'yes' ||  // Old Chrome, Opera etc.
+      window.navigator.msDoNotTrack === '1'  // IE
+  );
+}
+
+
+function serializeArgs(params) {
+  return Object.keys(params)
+    .filter(key => params[key] !== undefined)
+    .map(key => {
+      const val = params[key];
+      return encodeURIComponent(key) + (
+        val === null ? '' : '=' + encodeURIComponent(val));
+    })
+    .join('&');
+}
+
+
+class AmpDisqus extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    // API
+    this.preconnect.url('https://tempest.services.disqus.com/engage-iframe/amp/', onLayout);
+    // Image resources
+    this.preconnect.url('https://a.disquscdn.com', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.forum = user.assert(
```

Comment:
Please use trailing `_` for private properties everywhere. Mostly helps with minifying code.